File: da-test/src/test/java/ro/johann/da/test/api/service/decision/transfer/AlternativeOutput.java
package ro.johann.da.test.api.service.decision.transfer;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import lombok.Data;
import lombok.experimental.FieldDefaults;

import static lombok.AccessLevel.PRIVATE;

@Data
@FieldDefaults(level = PRIVATE)
@JsonIgnoreProperties(ignoreUnknown = true)
public class AlternativeOutput {
  String id;
  String name;
  Float utility;
  Integer rank;
}

File: da-decision-service/src/main/kotlin/ro/johann/da/decision/api/transfer/AlternativeOutput.kt
package ro.johann.da.decision.api.transfer

import com.fasterxml.jackson.annotation.JsonFormat
import ro.johann.da.decision.domain.Alternative
import ro.johann.da.decision.domain.ProcessedEntity
import java.time.LocalDateTime
import java.util.UUID

data class AlternativeOutput(
  val id: UUID,
  val name: String,
  val rank: Int? = null,
  val utility: Double? = null,
  @JsonFormat(pattern = "dd-MM-yyyy hh:mm:ss")
  val createdAt: LocalDateTime,
  @JsonFormat(pattern = "dd-MM-yyyy hh:mm:ss")
  val updatedAt: LocalDateTime
) {
  constructor(alternative: Alternative, processedEntity: ProcessedEntity? = null) : this(
    id = alternative.id,
    name = alternative.name,
    rank = processedEntity?.rank,
    utility = processedEntity?.utility,
    createdAt = alternative.createdAt,
    updatedAt = alternative.updatedAt
  )
}